### PR TITLE
Update to Python 3.10 (using repo branch)

### PR DESCRIPTION
**Pull request by jakirkham**
xref: https://github.com/conda-forge/webservices-dispatch-action/pull/19

### CODE REVIEW

**Comment by beckermr** [comment]
Ahhh this fails since I removed the branch from the test repo.

**Comment by jakirkham** [comment]
I see. The `rerender-live-test` is missing from [`cf-autotick-bot-test-package-feedstock`]( https://github.com/conda-forge/cf-autotick-bot-test-package-feedstock ), right? Can add that branch there. Was there anything special about that branch (`rerender-live-test`) or would just creating one adjacent to `main` be sufficient?

**Comment by jakirkham** [comment]
If you have a cloned copy of the repo, it may still have that remote branch (assuming its changes are important). Could also check `git reflog` or `git fsck`

**Comment by beckermr** [comment]
A branch on the repo will cause uploads. I think we need a branch from a fork.

**Comment by jakirkham** [comment]
That can be [overridden]( https://conda-forge.org/docs/maintainer/conda_forge_yml.html#upload-on-branch )

**Comment by beckermr** [comment]
That won't work here due to other tests run on this repo. Maybe the extra uploads are ok. They only happen once.

**Comment by jakirkham** [comment]
> That won't work here due to other tests run on this repo

Could you please clarify?

> Maybe the extra uploads are ok. They only happen once.

Was also wondering about this

<hr>

Any thoughts on this question?

> Was there anything special about that branch (`rerender-live-test`) or would just creating one adjacent to `main` be sufficient?

**Comment by beckermr** [comment]
We have automated tests of the webservices heroku app that require uploads on all branches. 

Our ci will only do uploads once so if we use a branch we only get one extra package. 

Afaik there was nothing special about that branch. 

Note that if we only test rerendering from a branch on the same repo then we could miss bugs from rerendering on forks.

**Comment by jakirkham** [comment]
Created `rerender-live-test` on [`cf-autotick-bot-test-package`]( https://github.com/conda-forge/cf-autotick-bot-test-package-feedstock ) and canceled CI to avoid package uploads

Have power cycled CI for this PR to see if that fixes the issue

File: Dockerfile [hunk 1]
```diff
@@ -40,7 +40,7 @@ RUN echo "**** install dev packages ****" && \
     mamba update --all && \
     mamba install --quiet \
         git \
-        python=3.8 \
+        python="3.10" \
         conda-smithy \
```
Comment (beckermr) [inline Dockerfile]:
```suggestion
        "conda-smithy>=3.22.0" \
```

File: tests/run_live_test.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def _run_test():
     r = requests.post(
         ("https://api.github.com/repos/conda-forge/"
          "cf-autotick-bot-test-package-feedstock/dispatches"),
-        data=json.dumps({"event_type": "rerender", "client_payload": {"pr": 19}}),
+        data=json.dumps({"event_type": "rerender", "client_payload": {"pr": 445}}),
```
Comment (jakirkham) [inline tests/run_live_test.py:59]: Ah ok. Now see what I was missing. Thanks Matt! 🙏

**Pull request merged**